libtransmission/handshake.cc

```diff
@@ -773,7 +772,9 @@ static ReadState readCryptoProvide(tr_handshake* handshake, tr_peerIo* peer_io)
 
     /* next part: ENCRYPT(VC, crypto_provide, len(PadC), */
 
-    peer_io->decryptInit(peer_io->isIncoming(), handshake->dh, *peer_io->torrentHash());
+    auto const& info_hash = peer_io->torrentHash();
+    TR_ASSERT_MSG(info_hash != tr_sha1_digest_t{}, "readCryptoProvide  requires an info_hash");
```

Comment:
nit: there are two consecutive spaces in this string.

Reply:
Fixed in 7863ed936a3cf597059f0eed7d2b72715413caf0